src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private Object[] elementData;
```

Comment:
we should use generic types for the fields this way we won't need to cast it to `T` on returns
```suggestion
    private T[] elementData;
```